haystack/components/evaluators/llm_evaluator.py

```diff
@@ -0,0 +1,230 @@
+import json
+from typing import Any, Dict, List, Optional, Tuple, Type
+
+from haystack import component, default_from_dict, default_to_dict
+from haystack.components.builders import PromptBuilder
+from haystack.components.generators import OpenAIGenerator
+from haystack.utils import Secret, deserialize_secrets_inplace
+
+
+@component
+class LLMEvaluator:
+    """
+    Uses an LLM to evaluate inputs based on a prompt containing instructions and examples.
+
+    The default API requires an OpenAI API key to be provided as an environment variable "OPENAI_API_KEY".
+    The inputs are lists that are user-defined depending on the desired metric.
+    The output is a dictionary with a key `results` containing a list of evaluation results.
+    Each result is a dictionary with user-defined keys and values of either 0 for FALSE or 1 for TRUE respectively.
+
+    Usage example:
+    ```python
+    from haystack.components.evaluators import LLMEvaluator
+    evaluator = LLMEvaluator(
+        instructions="Is this answer problematic for children?",
+        inputs=[("responses", List[str])],
+        outputs=["score"],
+    )
+    RESPONSES = [
+        "Football is the most popular sport with around 4 billion followers worldwide",
+        "Python language was created by Guido van Rossum.",
+    ]
+    results = evaluator.run(responses=RESPONSES)
+    ```
+    """
+
+    def __init__(
+        self,
+        instructions: str,
+        inputs: List[Tuple[str, Type]],
+        *,
+        outputs: Optional[List[str]] = None,
+        api: str = "openai",
+        api_key: Secret = Secret.from_env_var("OPENAI_API_KEY"),
+        examples: Optional[List[Dict[str, Any]]] = None,
+    ):
+        """
+        Creates an instance of LLMEvaluator.
+
+        :param instructions:
+            The prompt instructions to use for evaluation.
+            Should be a question about the inputs that can be answered with yes or no.
+        :param inputs:
+            The inputs that the component expects as incoming connections and that it evaluates.
+            Each input is a tuple of an input name and input type. Input types must be lists.
+        :param outputs:
+            Optional output names of the evaluation results. They correspond to keys in the output dictionary.
+            The default is a single key "score".
+        :param api:
+            The API to use for calling an LLM through a Generator.
+            Supported APIs: "openai".
+        :param api_key:
+            The API key.
+        :param examples:
+            Optional few-shot examples conforming to the expected input and output format as defined in the `inputs` and
+             `outputs` parameters.
+            Each example is a dictionary with keys "inputs" and "outputs"
+            They contain the input and output as dictionaries respectively.
+        """
+        self.instructions = instructions
+        self.inputs = inputs
+        self.outputs = outputs or ["score"]
```

Comment:
```suggestion
        self.outputs = outputs
```
See above. We also need some basic validation here.